### DISCUSSION TITLE
randomization.sampleWithoutReplacement(myArray, 1) must be array error despite array?

### DISCUSSION THREAD

**Discussion by nrschoen** [jspsych/jsPsych#1220] [Q&A] [answered]
I am trying to sample randomly from an array of strings and I am getting the "must be array" error even though the argument I insert is an array.

Here you can see my usage:
![Screen Shot 2020-11-23 at 14 04 45](https://user-images.githubusercontent.com/65179042/99965304-f5f7ce80-2d94-11eb-8b0a-d89fb69bb900.png)

Here the output with the console.log check indicating that "words" is indeed an array + the error message:
![Screen Shot 2020-11-23 at 13 57 01](https://user-images.githubusercontent.com/65179042/99965079-a0232680-2d94-11eb-8bb3-c7f0e1149366.png)

**Comment by jodeleeuw** [comment, marked as answer]
Hi @nrschoen ,

Because you are declaring `words` as the parameter name to the `on_load` function the variable has the value of `undefined`. The `on_load` function doesn't have any parameters passed when it is called. By putting the variable name in the function parameters it is overwritten within the scope of the function.

Try removing `words` as a parameter, so that you can access the global variable `words` instead.

```js
on_load: function(){
	var current_word = ...
}
```

**Reply by nrschoen** [reply]
Thank you! This was it :)
